api/v1beta1/glance_webhook.go

```diff
@@ -194,6 +194,11 @@ func (r *GlanceSpecCore) isInvalidBackend(glanceAPI GlanceAPITemplate, topLevel
 	if *glanceAPI.Replicas != rep && glanceAPI.Type == "split" && isFileBackend(glanceAPI.CustomServiceConfig, topLevel) {
 		return true
 	}
+	// Do not allow to deploy a glanceAPI with a backend other than File
+	// (Cinder, Swift, Ceph) with "type: single": we must split in that case
+	if glanceAPI.Type == APISingle && !isFileBackend(glanceAPI.CustomServiceConfig, topLevel) {
```

Comment:
@konan-abhi I probably need to have here `*glanceAPI.Replicas != rep` as well, to not fail when the backend is still not available (and the human operator put `replicas: 0`)